src/dataSources/cloudFirestore/community.js

```diff
@@ -167,11 +222,67 @@ const community = dbInstance => {
     });
 
     await communityDocRef.update(moddedCommunity);
-    const updatedDoc = await communityDocRef.get();
-    return {
-      id: updatedDoc.id,
-      ...updatedDoc.data(),
-    };
+
+    return get(communityDocRef.id);
+  }
+
+  async function changeSlug({ communityId, newSlug, user }) {
+    dlog(
+      'change community slug called for: %s, newSlug: %s',
+      communityId,
+      newSlug,
+    );
+    const isNewInUse = await slugStore(dbInstance).isSlugTaken(newSlug);
+    if (isNewInUse)
+      throw new Error(
+        'unable to change communuity slug, new slug in use already',
+      );
+    const communityDocRef = communityCol.doc(communityId);
+    const docSnapshot = await communityDocRef.get();
+    if (!docSnapshot.exists)
+      throw new Error(
+        'invalid communityId provided, unable to change community slug',
+      );
+    const cleanCommunity = scrubCommunity({
+      community: {
+        slug: newSlug,
+      },
+      user,
+    });
+    const currentSlug = docSnapshot.get('slug');
+    const currentSlugDocRef = slugStore(dbInstance).getSlugDocRef(currentSlug);
```

Comment:
any reason you don't batch these three into a promise.all?

Reply:
getSlugRef doesn't return a Promise. It is a reference created by the sdk. This reference is needed for the batch write to Firestore